### PR TITLE
Add JS modules

### PR DESCRIPTION
**Pull request by foolip**

### CODE REVIEW

File: feature-group-definitions/js-modules.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    firefox_android: "60"
+    safari: "10.1"
+    safari_ios: "10.3"
+compat_features:
+  - html.elements.script.nomodule
+  - html.elements.script.type.module
+  - javascript.statements.export
+  - javascript.statements.export.default
+  - javascript.statements.import
+  - svg.elements.script.type.module
```
Comment (foolip) [inline feature-group-definitions/js-modules.yml:30]: https://github.com/web-platform-dx/web-features/pull/677 removes this from another feature.

**Review by ddbeck — APPROVED** [review]
Approving, though I'd be greatly reassured if @Elchi3 wanted to take a look at this. 😄

**Review by Elchi3 — COMMENTED** [review]
I would have said make it a bundle (😄) called "JavaScript modules (initial support)" and then create a group named "JavaScript Modules", and introduce more bundles belonging to that group. 

For example:

- Group: JavaScript modules
  - Bundle: JavaScript modules (initial support) 
  - Bundle: JavaScript modules in SVG
  - Bundle: JavaScript modules in Workers  
  - Bundle: JavaScript modules in ServiceWorkers
  - Bundle: JavaScript modules in SharedWorkers

**Comment by foolip** [comment]
@Elchi3 Good suggestion, an initial support variant makes sense here.

This runs into something I've been anticipating: should we use `js-modules` as the identifier for both the group and the initial feature? I think we probably should, and the consequence is that groups and features will be distinct namespaces and consumers need more than a bare identifier to refer to a group.

**Comment by Elchi3** [comment]
To me the "JS module" group acts as an umbrella for all things that have to do with modules on a technical level. And I think for exploring the web platform, these technical groups are interesting.

I can imagine that "JS modules (initial support)" becomes part of other groups and "JS Modules in SVG" doesn't.

For example, https://www.webcomponents.org/introduction mentions that JS modules also belong to set of web platform APIs that make "Web Components".

Group: Web Components
- JS modules (initial support) (Consumers can decide to provide the option to zoom in to see the all other features grouped "JS module", in case you really want to do something special.)
- Shadow DOM (same)
- Custom Elements (same)

**Comment by foolip** [comment]
I've pushed a change to remove `svg.elements.script.type.module` from the other feature so that this PR makes sense on its own.

I'll let this sit for a few days (a.k.a. weekend) to collect my thoughts on how best to handle the evolution of this feature.

**Comment by ddbeck** [comment]
I brought this up with @Fyrd on a call today and here are some things I learned:

I asked him where he might draw the line on making distinct features for feature evolution, in view of making a feature on caniuse or consuming one from web-features. After a little bit of discussion, I linked to https://github.com/web-platform-dx/web-features/pull/676#pullrequestreview-1925024614 and asked him whether he would group those things together (perhaps with an indicator of "partial implementation" until all of the pieces were satisfied) or split them. His response was, almost word for word: "Why not both?" 😄 

When I pressed him a bit on how he'd decide without the help of web-features, he said that he would probably err on the side of a larger feature until (paraphrasing) 'it became a problem'. That is, he'd lump things together unless he had folks asking him for finer detail. He talked a little bit about developers asking about certain details of a given feature's support and not necessarily learning the answer until after the question was raised. Sometimes that leads to changes in what features are covered on caniuse.

But on the "both" line of thinking, I think our snapshots (and probably groups) approach was very much validated: Alexis linked to https://caniuse.com/?search=ecmascript%20202, to show that he has implemented some bespoke searches on caniuse that do what is already suggested by snapshots. Heretofore unknown to me, caniuse users sometimes use the site in an exploratory way, to find out what's new in the web platform. Alexis has responded by surfacing categories, instead of trying to track support for large, only-distantly related sets of features.

To editorialize briefly, to me this suggests two things to explore:

- A principle of parsimony: unless we have evidence that developers are mentally splitting (features/bundles or groups), we ought to prefer more comprehensive features and groups.
- Considering the possibility of composing features (e.g., to show the status of feature A + feature B) for our authorial convenience, but deferring to consumers to decide whether to always, sometimes, or never show the composed or decomposed features.

**Comment by foolip** [comment]
Thank out @ddbeck and @Fyrd, that sounds great to me. I'm not sure how to implement it in this PR yet, but I like the direction.

**Comment by ddbeck** [comment]
For the purposes of this PR, the answer might just be to merge it (unless @Elchi3 thinks the evolutionary bundles are strictly necessary, in addition to the general feature shown here)? We can add more, finer-sliced features later, if they turn out to be distinct/relevant to developers.

**Pull request merged**